samples/nrf9160/modem_shell/src/drivers/nrf91_non_offloaded_dev.c

```diff
@@ -0,0 +1,507 @@
+	struct in_addr dns_addr4_secondary;
+	struct in6_addr dns_addr6_primary;
+	struct in6_addr dns_addr6_secondary;
+	uint32_t ipv4_mtu;
+	uint32_t ipv6_mtu;
+};
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO "AT+CGCONTRDP=%d"
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO_PARAM_COUNT 20
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO_DNS_ADDR_PRIMARY_INDEX 6
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO_DNS_ADDR_SECONDARY_INDEX 7
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO_MTU_INDEX 12
+
+static int util_get_pdn_conn_dyn_params(int cid, struct pdn_conn_dyn_params *ret_dyn_info)
+{
+	struct at_param_list param_list = { 0 };
+	size_t param_str_len;
+	char *next_param_str;
+	char at_cmd_str[16];
+	char at_cmd_response_str[1024];
+	char dns_addr_str[256];
+	char *at_ptr = at_cmd_response_str;
+	int ret, family, iterator;
+	struct in_addr *addr;
+	struct in6_addr *addr6;
+	bool resp_continues = true;
+
+	sprintf(at_cmd_str, AT_CMD_PDP_CONTEXT_READ_DYN_INFO, cid);
+
+	ret = nrf_modem_at_cmd(at_cmd_response_str, sizeof(at_cmd_response_str), at_cmd_str);
+	if (ret) {
+		mosh_error("Cannot get PDP conn dyn params, ret: %d", ret);
+		return false;
+	}
+
+	/* Parse the response */
+	ret = at_params_list_init(&param_list, AT_CMD_PDP_CONTEXT_READ_DYN_INFO_PARAM_COUNT);
+	if (ret) {
+		mosh_error("Could not init AT params list, error: %d\n", ret);
+		return ret;
+	}
+
+	iterator = 0;
+	while (resp_continues) {
+		resp_continues = false;
+		ret = at_parser_max_params_from_str(at_ptr, &next_param_str, &param_list,
+						    AT_CMD_PDP_CONTEXT_READ_DYN_INFO_PARAM_COUNT);
+		if (ret == -EAGAIN) {
+			resp_continues = true;
+		} else if (ret == -E2BIG) {
+			mosh_error("E2BIG, error: %d\n", ret);
+		} else if (ret != 0) {
+			mosh_error("Could not parse AT response for %s, error: %d\n", at_cmd_str,
+				   ret);
+			goto clean_exit;
+		}
+
+		/* Read primary DNS address */
+		param_str_len = sizeof(dns_addr_str);
+		ret = at_params_string_get(&param_list,
+					   AT_CMD_PDP_CONTEXT_READ_DYN_INFO_DNS_ADDR_PRIMARY_INDEX,
+					   dns_addr_str, &param_str_len);
+		if (ret) {
+			mosh_error("Could not parse dns str for cid %d, err: %d", cid, ret);
+			goto clean_exit;
+		}
+		dns_addr_str[param_str_len] = '\0';
+
+		if (dns_addr_str != NULL) {
+			family = net_utils_sa_family_from_ip_string(dns_addr_str);
+
+			if (family == AF_INET) {
+				addr = &(ret_dyn_info->dns_addr4_primary);
+				(void)inet_pton(AF_INET, dns_addr_str, addr);
+			} else if (family == AF_INET6) {
+				addr6 = &(ret_dyn_info->dns_addr6_primary);
+				(void)inet_pton(AF_INET6, dns_addr_str, addr6);
+			}
+		}
+
+		/* Read secondary DNS address */
+		param_str_len = sizeof(dns_addr_str);
+
+		ret = at_params_string_get(
+			&param_list, AT_CMD_PDP_CONTEXT_READ_DYN_INFO_DNS_ADDR_SECONDARY_INDEX,
+			dns_addr_str, &param_str_len);
+		if (ret) {
+			mosh_error("Could not parse dns str, err: %d", ret);
+			goto clean_exit;
+		}
+		dns_addr_str[param_str_len] = '\0';
+
+		if (dns_addr_str != NULL) {
```

Comment:
Same as above, this is an array from the stack and can not be NULL, so the check can be removed.

Reply:
Yep, this check was removed.